Orange/widgets/utils/plot/owplotgui.py

```diff
@@ -27,18 +27,225 @@
+        add_remove.layout().addStretch(10)
+        add_remove.layout().setSpacing(1)
+        add_remove.setSizePolicy(*SIZE_POLICY_FIXED)
+        master.controlArea.layout().addWidget(add_remove)
+        self.add_remove = add_remove
+        self.box = add_remove.buttons[1]
+
+        self.varmodel_other = model = DomainModel(**model_params)
+
+        self.other_view = view = gui.listView(None, master, box="Hidden Axes", **params_view)
+        view.setMinimumSize(QSize(30, 60))
+        view.setSizePolicy(*SIZE_POLICY_ADAPTING)
+        view.viewport().setAcceptDrops(True)
+        moveup = QAction(
+            "Move up", view,
+            shortcut=QKeySequence(Qt.Key_Delete),
+            triggered=self.__activate_selection
+        )
+        view.addAction(moveup)
+
+        view.setModel(model)
+        self.filter_edit, self.other_view = variables_filter(
+            model=model)
+        self.hidden = Hidden(self)
+
+        return self.varmodel_selected, self.varmodel_other
+
+    def set_enabled(self, is_enabled):
+        self.active_view.setEnabled(is_enabled)
+        self.other_view.setEnabled(is_enabled)
+        self.hidden.btn_select.setEnabled(is_enabled)
+        self.hidden.btn_cancel.setEnabled(is_enabled)
+        for btn in self.add_remove.buttons:
+            btn.setEnabled(is_enabled)
+        self.filter_edit.setEnabled(is_enabled)
+
+    def display_select(self):
+        view = self.other_view
+        model = self.varmodel_other
+        indices = view.selectionModel().selectedRows()
+
+        variables = [model.data(ind, Qt.EditRole) for ind in indices]
+
+        for i in sorted((ind.row() for ind in indices), reverse=True):
+            del model[i]
+
+        self.varmodel_selected.extend(variables)
+
+    def display_all(self):
+        self.varmodel_selected[:] += self.varmodel_other[:]
+        self.varmodel_other[:] = []
+
+    def display_none(self):
+        self.varmodel_other[:] += self.varmodel_selected[:]
+        self.varmodel_selected[:] = []
+
+    def __activate_selection(self):
+        view = self.other_view
+        model = self.varmodel_other
+        indices = view.selectionModel().selectedRows()
+
+        variables = [model.data(ind, Qt.EditRole) for ind in indices]
+
+        for i in sorted((ind.row() for ind in indices), reverse=True):
+            del model[i]
+
+        self.varmodel_selected.extend(variables)
+
+    def __deactivate_selection(self):
+        view = self.active_view
+        model = self.varmodel_selected
+        indices = view.selectionModel().selectedRows()
+
+        variables = [model.data(ind, Qt.EditRole) for ind in indices]
+
+        for i in sorted((ind.row() for ind in indices), reverse=True):
+            del model[i]
+
+        self.varmodel_other.extend(variables)
+
+    def _action_add(self):
+        self.hidden(self)
+
+    @staticmethod
+    def encode_var_state(lists):
```

Comment:
Are these encodes and decodes really needed? Cannot settings take care of lists of variables? Is this again some legacy from the previous form of the widget?

Reply:
Yes, they are needed.